_includes/related-posts.html

```diff
@@ -0,0 +1,43 @@
+<section class="usa-grid usa-section posts_feature">
+  {% assign hasSimilar = '' %}
+  {% for post in site.related_posts %}
+    {% assign postHasSimilar = false %}
+    {% for tag in post.tags %}
+      {% for thisTag in page.tags %}
+        {% if postHasSimilar == false and hasSimilar.size < 3 and post != page and tag == thisTag %}
+          {% if hasSimilar.size == 0 %}
+          <h2>Related Posts</h4>
+          <ul>
+          {% endif %}
+          <li class="usa-width-one-third">
+            <article>
+              <p class="post-date">{{ post.date | date: "%B %-d, %Y" }}</p>
+              <h3 class="posts_feature-heading">
+                <a href="{{ site.baseurl }}{{ post.url }}">{{ post.title }}
+                {% if post.series %}
+                  (Series: {{ post.series }})
+                {% endif %}
+                </a>
+              </h3>
+              <p>{{ post.excerpt }}</p>
+              <p>
+                <a class="link-arrow-right post-link-continue_reading" href="{{ post.url | prepend: site.baseurl }}">Continue reading</a>
+              </p>
+              <span class="post-tags" itemprop="keywords">
+                {% for tag in post.tags %}
+                  <a class="usa-label" href="{{ site.baseurl }}/{{ site.tag_dir }}/{{ tag | slugify }}/">{{ tag }}
+                  </a>
+                {% endfor %}
+              </span>
+            </article>
+          </li>
+          {% capture hasSimilar %}{{ hasSimilar }}*{% endcapture %}
```

Comment:
Interested to know more about this. Seems like `hasSimilar` is capturing itself?